hypervisor/src/mshv/mod.rs

```diff
@@ -611,6 +617,392 @@ impl cpu::Vcpu for MshvVcpu {
+                                                [(
+                                                hv_register_name_HV_X64_REGISTER_SEV_DOORBELL_GPA,
+                                                ghcb_doorbell_gpa.as_uint64,
+                                            )]
+                                            };
+                                            set_registers_64!(self.fd, reg_names).map_err(|e| {
+                                                cpu::HypervisorCpuError::SetRegister(e.into())
+                                            })?;
+
+                                            let mut sw_exit_info2_gpa =
+                                                mshv_bindings::mshv_read_write_gpa {
+                                                    base_gpa: ghcb_gpa + GHCB_SW_EXITINFO2_OFFSET,
+                                                    byte_count: std::mem::size_of::<u64>() as u32,
+                                                    ..Default::default()
+                                                };
+                                            sw_exit_info2_gpa.data[0..8]
+                                                .copy_from_slice(&exit_info2.to_le_bytes());
+                                            self.fd.gpa_write(&mut sw_exit_info2_gpa).map_err(
+                                                |e| cpu::HypervisorCpuError::GpaWrite(e.into()),
+                                            )?;
+
+                                            // Clear the SW_EXIT_INFO1 register to indicate no error
+                                            let clear_sw_exit_info1 = 0_u64;
+                                            let mut sw_exit_info1_gpa =
+                                                mshv_bindings::mshv_read_write_gpa {
+                                                    base_gpa: ghcb_gpa + GHCB_SW_EXITINFO1_OFFSET,
+                                                    byte_count: std::mem::size_of::<u64>() as u32,
+                                                    ..Default::default()
+                                                };
+                                            sw_exit_info1_gpa.data[0..8].copy_from_slice(
+                                                &clear_sw_exit_info1.to_le_bytes(),
+                                            );
+                                            self.fd.gpa_write(&mut sw_exit_info1_gpa).map_err(
+                                                |e| cpu::HypervisorCpuError::GpaWrite(e.into()),
+                                            )?;
+                                        }
+                                        SVM_NAE_HV_DOORBELL_PAGE_QUERY => {
+                                            let reg_names =
+                                                [hv_register_name_HV_X64_REGISTER_SEV_DOORBELL_GPA];
+                                            let mut reg_assocs: Vec<hv_register_assoc> = reg_names
```

Comment:
There's no need to create a Vec here, is there? Just put the `hv_register_assoc`s in an array on the stack.

Reply:
Correct fixed in the new revision.